src/test/resources/insert/statement/new_revision_paths

```diff
@@ -0,0 +1,2 @@
+#containedType[name=Statement]#method[signature=main(java.lang.String[])]#body#statement[name=b]
```

Comment:
Why do some of these use `statement[name=something]` while others use `statement[index=something]`? It seems to me like they should all use `index=something`, I don't think the `name=something` encodes the statement's position in any way (and so, it can't be verified).

Reply:
I got these paths by directly querying `getPath` from new revision nodes.

Reply:
Change them manually to use statement index.

Reply:
I haven't tried it out yet but it might pose problem because `pathString` will be computed as `#containedType[name=Statement]#method[signature=main(java.lang.String[])]#body#statement[name=b]` and **not** `#containedType[name=Statement]#method[signature=main(java.lang.String[])]#body#statement[name=1]` so out test might try to assert this line from the `prev` file.

Reply:
Ah, that is a good point. I didn't think of that.

Perhaps we must keep them as they are generated, then.

Reply:
Actually, I had a quick look and know how it can be solved.

Before executing, just evaluate all of the paths to get the elements. Then, instead of comparing paths, you just compare the elements directly.

Reply:
Sure. But don't you think we should keep the way they are generated even though it will be a little inconsistent? The reason for that is we won't need to pay much attention to creating metadata of each test case and just trust that `getPath` works correctly.

Reply:
Well, we can, but it weakens the test as the path doesn't refer to a specific index (and we very much care about specific indices).

Then again, the original test should verify that anyway, so it's not too much of a problem.

On a related note: these should really be two different tests (one to check structure, one to check element origin). Otherwise, when a test fails, you won't immediately know if it's due to incorrect structure, or incorrect element origin. But you can refactor that at a later time if you're in a hurry now.

Reply:
> Well, we can, but it weakens the test as the path doesn't refer to a specific index (and we very much care about specific indices).

Yeah, you're right about that. I think this should be reported to spoon. Although the statement in block can be uniquely identified with the variable name, if Spoon uses index, it should consistently use that.

Anyway, I will rewrite the if condition so it checks for element equality rather than path equality because we are essentially checking for element origin.

> But you can refactor that at a later time if you're in a hurry now.

Yes, I will open an issue about it once we merge this PR and tackle that later.

Reply:
> Before executing, just evaluate all of the paths to get the elements. Then, instead of comparing paths, you just compare the elements directly.

We have access to the paths in metadata file but they are just `String`s. How can I use them to evaluate the path on the new package?

Reply:
```java
CtPath path = new CtPathStringBuilder().fromString(pathString);
```